.circleci/config.yml

```diff
@@ -21,6 +21,43 @@ jobs:
             dep check
 
 
+  lint:
+    docker:
+      - image: circleci/golang:1.14
+    environment:
+      GOPATH: /home/circleci/.go_workspace
+      GOLANGCI_VERSION: v1.20.0
+    working_directory: /home/circleci/.go_workspace/src/github.com/loadimpact/k6
+    steps:
+      # Workaround for custom env vars not available in cache keys
+      # https://discuss.circleci.com/t/cannot-use-circle-yml-environment-variables-in-cache-keys/10994/8
+      - run: echo $GOLANGCI_VERSION > /tmp/.golangci.version
+      - restore_cache:
+          name: Restore golangci-lint cache
+          key: golangci-lint-{{ arch }}-{{ checksum "/tmp/.golangci.version" }}-{{ .Environment.CACHE_VERSION }}
+          paths:
+            - /home/circleci/.go_workspace/bin/golangci-lint
+      - run:
+          name: Install golangci-lint
+          command: |
+            export PATH="$GOPATH/bin:$PATH"
+            command -v golangci-lint && exit
+            GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@$GOLANGCI_VERSION
+      - save_cache:
+          name: Save golangci-lint cache
+          key: golangci-lint-{{ arch }}-{{ checksum "/tmp/.golangci.version" }}-{{ .Environment.CACHE_VERSION }}
+          paths:
+            - /home/circleci/.go_workspace/bin/golangci-lint
+      - checkout
+      - run:
+          name: Run golangci-lint
+          command: |
+            export PATH="$GOPATH/bin:$PATH"
+            basecommit=$(git log --first-parent --pretty=format:'%H %D' | egrep '/(master|new-schedulers)' | head -1 | cut -d' ' -f1)
+            echo "basecommit=$basecommit"
+            golangci-lint run --out-format=tab --new-from-rev "$basecommit" ./...
```

Comment:
I thought a variable base commit would be more useful than having `--new-from-rev master` hard-coded here, at least until `new-schedulers` is merged. I ran into some issues with CircleCI as it doesn't expose the parent branch (see 22a2e17f), which to be fair doesn't make much sense in Git anyway.

Reply:
To be fair, we want to merge `new-schedulers` in `master` anyway, and that will likely be true for other long-running branches... So it probably would be ok if we only ever compare with `master`

Reply:
Indeed... Should I leave it as `--new-from-rev master` then?

Reply:
hmm leave it as it is for now, until we move to github actions - given how many changes there are in `new-schedulers` compared to `master`, it wouldn't be out of the question for the determination to be wrong - this is probably more accurate for PRs to `new-schedulers`

Reply:
I think the "correct" way to do it is:
1. if what you are building is a PR - compare it to where it will be merged into 
2. if this is just a commit on branch - compare it to the previous commit on this branch ... (this shadows old errors which is problematic)
3. if master (special case for special branch) - compare to the commit we started using golangci from ... as we don't want to shadow those 

By quick look at what [circleci exposes as variable](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables) this will be .. hard so I propose to just leave it and try to it "correct"ly in github actions

Reply:
Yeah, this needs to be smarter. Right now commits on `master` or `new-schedulers` will silently pass, since it's effectively checking against `HEAD` :-/

We'll probably need to refine this before the move to Actions.